vm/systemSmartContracts/delegation.go

```diff
@@ -929,6 +935,30 @@ func (d *delegation) checkBLSKeysIfExistsInStakingSC(blsKeys [][]byte) bool {
 	return false
 }
 
+func (d *delegation) changeOwner(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
+	if !d.flagChangeDelegationOwner.IsSet() {
+		d.eei.AddReturnMessage(args.Function + " is an unknown function")
+		return vmcommon.UserError
+	}
+
+	returnCode := d.checkOwnerCallValueGasAndDuplicates(args)
+	if returnCode != vmcommon.Ok {
+		return returnCode
+	}
+	if len(args.Arguments) != 1 {
+		d.eei.AddReturnMessage("wrong number of arguments, expected 1")
+		return vmcommon.UserError
+	}
+	if len(args.Arguments[0]) != len(args.CallerAddr) {
+		d.eei.AddReturnMessage("invalid argument, wanted an address")
+		return vmcommon.UserError
+	}
+
+	d.eei.SetStorage([]byte(ownerKey), args.Arguments[0])
```

Comment:
what happens to the initialOwnerFunds, 1250 EGLD, of the previous owner? When can he withdraw?

Reply:
he cannot withdraw - that is for the new owner.

Reply:
ok... so during the offchain agreement to change the owner, the new owner should reimburse the old owner the initialOwnerFunds?